mvebu: add Turris Omnia LED support

kkudielka opened the pull request:
This pull request was originally split into three commits. I have now squashed them into a single commit, to be more compliant with the guidelines. The commit message contains all the details.

JamesT42 commented on target/linux/mvebu/base-files/etc/config/omnia:
```diff
@@ -0,0 +1,41 @@
+config led
+	option sysfs 'omnia-led:pci3'
+	option autonomous '1'
+
```
For convenience it would be nice to have every led in here, that would ease setting color because one would not have to lookup the exact name.

kkudielka replied:
Done.

JamesT42 commented:
I tested this and found no errors!

kkudielka committed:
mvebu: add Turris Omnia LED support

This patch adds LED support for the Turris Omnia, and UCI configuration
for the non-standard features (color, autonomous, global_brightness).

target/linux/mvebu/patches-4.14/490-omnia-leds.patch:
- Add the kernel driver of TurrisOS.
  (https://gitlab.labs.nic.cz/turris/turris-build/commit/60f75b6b)
- Add LED definitions to the Turris Omnia device tree.

package/kernel/linux/modules/i2c.mk:
- Make i2c-mux-pca954x load early, otherwise devices behind the MUX
  (such as the LEDs) are not found.

package/kernel/linux/modules/leds.mk:
- Add KernelPackage/leds-omnia.
- Make leds-omnia load early, to enable LED diagnostics during boot.

target/linux/mvebu/image/cortex-a9.mk:
- Add the necessary kmod package to the Turris Omnia image.

target/linux/mvebu/base-files/etc/diag.sh:
- Make omnia-led:power the status LED.

target/linux/mvebu/base-files/lib/preinit/09_initialize_leds:
- Add a preinit hook, to clear the "autonomous" flag for all LEDs.
  Otherwise, the LEDs would be controlled by hardware, and LED diagnostics
  would not work.

target/linux/mvebu/base-files/etc/board.d/01_leds:
- Generate default configuration for non-autonomous LEDs.
- Let the WAN LED use the netdev trigger (also works with SFP).

target/linux/mvebu/base-files/etc/init.d/omnia_led:
- Parse configuration for all extra features of the Turris Omnia LED
  driver (global brightness, individual LED color, individual autonomous
  flag).
- If no configuration exists, generate one with sane defaults.

To apply changes to the "omnia_led" config, users should execute
"service omnia_led start".

Tested on a Turris Omnia.

Signed-off-by: Klaus Kudielka <klaus.kudielka@gmail.com>
